sts-mock/src/token/keyRetriever/keyRetriever.ts

```diff
@@ -0,0 +1,76 @@
+import { errorResult, Result, successResult } from "../../utils/result";
+import { GetObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { createPrivateKey, KeyObject, JsonWebKey } from "node:crypto";
+
+export type SigningKey = {
+  signingKey: KeyObject;
+  keyId: string;
+};
+
+type PrivateKeyJwk = JsonWebKey & {
+  kid: string;
+  kty: "EC";
+  crv: "P-256";
+  d: string;
+  x: string;
+  y: string;
+};
+
+export interface IKeyRetriever {
+  getKey: (bucketName: string, fileName: string) => Promise<Result<SigningKey>>;
+}
+
+export class KeyRetriever implements IKeyRetriever {
+  private s3Client = new S3Client([
+    {
+      region: "eu-west-2",
+      maxAttempts: 2,
+    },
+  ]);
+
+  async getKey(
+    bucketName: string,
+    fileName: string,
+  ): Promise<Result<SigningKey>> {
+    const commandInput = {
+      Bucket: bucketName,
+      Key: fileName,
+    };
+
+    let key;
+    try {
+      const response = await this.s3Client.send(
+        new GetObjectCommand(commandInput),
+      );
+      key = await response.Body!.transformToString();
```

Comment:
Should we look at handling the scenario where the body doesn't come through? I'm not too keen on using the `!`

Reply:
I think let's put our best foot forward unless we need to consciously make a decision to reduce quality to reduce engineering effort. It's a good question to ask and one that we could debate :D

Reply:
After doing some research, it seems that `transformToString()` will throw an error if `Body` is falsy, but as this line of code is in the catch block, that error will be caught. However, I have now added an if statement to check if `Body` is falsy and return a different error message if it is. Please have a look. 🥺